lib/features/mailbox_dashboard/domain/repository/email_cache_repository.dart

```diff
@@ -0,0 +1,10 @@
+import 'package:jmap_dart_client/jmap/mail/email/email.dart';
+import 'package:tmail_ui_user/features/mailbox_dashboard/data/model/email_cache.dart';
+
+abstract class EmailCacheRepository {
+  void saveEmailCacheOnWeb(Email email);
+
+  EmailCache? getEmailCacheOnWeb();
```

Comment:
why not a future?

Reply:
I am not a fan of null result. I prefer the value if we can get it, otherwise throw an exception.